### ISSUE TITLE
Do not use try and catch as flow control

### ISSUE THREAD

**Issue by jowr** [CoolProp/CoolProp#877]
There are quite a few cases where try/catch blocks are used to control the computational flow. For example in the tabular backends (try loading - fall back to generate tables) and in the HelmholtzEOSMixtureBackend (try Newton solver, fall back to Secant, fall back to Brent). 

There are a few issues here in terms of speed (exception handling tends to be slow) and in terms of debugging (I got confused while hunting a division by zero bug and there were many exceptions around).

**Comment by ibell** [comment]
Yeah, I can see the point on both fronts.  I'm not sure how to architect
this to make this part nicer, but I am sure there is a way.  I am certainly
open to suggestions (and PR).

Is there a measureable speed penalty for using the exceptions?

**Comment by jowr** [comment]
I think that we can keep this issue as some sort of a collection for all the problems related to exceptions in the DLL (#879, #891, #892) when calling from Delphi. I am looking forward to merge #893 , but I think the root of the problems is the excessive use of try and catch blocks which takes the risks of running into invalid floating point operations. 

I have no numbers for the associated speed penalties, but the links that @bru32 posted earlier (http://www.virtualdub.org/blog/pivot/entry.php?id=53) suggest that there is potential for acceleration. Let me quote it here: 

> In general, you don't want to be tripping floating-point exceptions, even if they are masked. The reason is that when the FPU hits one, the fast hardware can't handle it and punts to the microcode, which then takes about twenty times longer. This is especially bad with NaNs since any operation with a NaN produces another NaN, causing them to spread throughout your calculations (NaN disease) and slow down everything massively. You can even crash due to NaNs blowing past clamp expressions, since any comparison with a NaN is false and converting one to integer form results in integer indefinite (0x80000000). Despite the erroneous results, though, NaNs can appear sporadically in a large Win32 program without anyone knowing, and may go unnoticed in a code base for years.

**Comment by ibell** [comment]
I think the first target should be the numerical solvers.  I propose that we implement the solvers such that they themselves do not throw, but rather they return an error code (a la REFPROP) that is 0 if fine,  >0 if error, or <0 if warning.  Not sure how to handle the functions called by the solvers.

**Comment by ibell** [comment]
Closing as part of an issue-tracker cleanup — this issue has had no activity in over two years.

*To be transparent: this comment and the close action are being performed by Claude (Anthropic's AI coding assistant), running on behalf of @ibell. Ian reviewed and approved the bulk-close criteria; I am the one executing the action.*

If you would like this issue reconsidered, please feel free to **re-open it** — a fresh comment with current behaviour against the latest CoolProp release is the most useful way to revive the discussion.

— Claude (acting for @ibell)
